js/pages/settings.js

```diff
@@ -173,44 +191,45 @@ const handleEditNameSection = () => {
       middleName = middleNameField.value.trim();
       isNameFormDisplayed = toggleElementVisibility(nameElementArray, isNameFormDisplayed);
       toggleButtonText();
-
-      changeName(firstName, lastName, middleName, suffix, preferredFirstName)
-        .then(() => {
-          handleOptionalFieldVisibility(middleName, 'profileMiddleName', middleNameRow, nameElementArray[0], 'text');
-          handleOptionalFieldVisibility(suffix, 'profileSuffix', suffixRow, nameElementArray[0], 'suffix');
-          handleOptionalFieldVisibility(preferredFirstName, 'profilePreferredFirstName', preferredFirstNameRow, nameElementArray[0], 'text');
-          successMessageElement = document.getElementById('changeNameSuccess');
-          successMessageElement.style.display = 'block';
-          document.getElementById('profileFirstName').textContent = firstName;
-          document.getElementById('profileLastName').textContent = lastName;
-        })
-        .catch(function (error) {
-          document.getElementById('changeNameFail').style.display = 'block';
-          document.getElementById('changeNameError').innerHTML = error.message;
-        });
+      submitNewName(firstName, lastName);
     }
   });
 };
 
+const submitNewName = async (firstName, lastName) => {
+  const isSuccess = await changeName(firstName, lastName, middleName, suffix, preferredFirstName, userData).catch(function (error) {
```

Comment:
is there a reason why some fields you're having global constants (i.e. middle name) and others you are declaring as needed? (i.e. first name)

Reply:
Yes - required fields are declared as needed because we know they will be present, so the scope is narrow. Optional fields are declared as globals so they can be toggled based on how the user interacts with the form and what the user edits.